src/FieldWarning/Assets/Units/Component/Weapon/IWeapon.cs

```diff
@@ -28,8 +28,8 @@ public interface IWeapon
         /// </summary>
         /// <param name="target"></param>
         /// <param name="deltaTime">Time since the last invocation.</param>
-        /// <param name="distanceToTarget">Distance from the firing unit to the target unit</param>
+        /// <param name="displacement">Distance from the firing unit to the target unit</param>
         /// <returns>True if a shot was fired, false otherwise.</returns>
-        bool TryShoot(TargetTuple target, float deltatime, float distanceToTarget);
+        bool TryShoot(TargetTuple target, float deltatime, Vector3 displacement);
```

Comment:
Displacement is not a fancy word for distance, it means something completely different :)

Reply:
That represents the variable is changed from a float value to a Vector3 value, seems perfectly fine since it is physically the displacement from firing unit to target (a vector quantity, not a scalar)

Reply:
I looked it up and I see that you're correct that there's a mathematical term called 'displacement' for vectors. I think it is not in common use and using it there is fairly confusing because of that, but you're technically correct so it is up to you.